src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,664 @@
+        for (uint256 i = 0; i < strategiesAndWeightingMultipliers.length - 1; i++) {
+            strategiesAndWeightingMultipliers1[i] = strategiesAndWeightingMultipliers[i];
+        }
+        strategiesAndWeightingMultipliers2[0] = strategiesAndWeightingMultipliers[strategiesAndWeightingMultipliers.length - 1];
+
+        // create quorum with all but the last element
+        uint8 quorumNumber = voteWeigher.quorumCount();
+        cheats.prank(serviceManagerOwner);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers1);
+
+        // add the last element
+        cheats.prank(notServiceManagerOwner);
+        cheats.expectRevert("VoteWeigherBase.onlyServiceManagerOwner: caller is not the owner of the serviceManager");
+        voteWeigher.addStrategiesConsideredAndMultipliers(quorumNumber, strategiesAndWeightingMultipliers2);
+    }
+
+    function testAddStrategiesConsideredAndMultipliers_ForNonExistantQuorum_Reverts(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers
+    ) public {
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+        // we need 2 or more strategies to create the quorums and then add one
+        cheats.assume(strategiesAndWeightingMultipliers.length > 1);
+
+        // separate strategiesAndWeightingMultipliers into 2 arrays, 1 with the last element and 1 with the rest
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers1 = new IVoteWeigher.StrategyAndWeightingMultiplier[](strategiesAndWeightingMultipliers.length - 1);
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers2 = new IVoteWeigher.StrategyAndWeightingMultiplier[](1);
+        for (uint256 i = 0; i < strategiesAndWeightingMultipliers.length - 1; i++) {
+            strategiesAndWeightingMultipliers1[i] = strategiesAndWeightingMultipliers[i];
+        }
+        strategiesAndWeightingMultipliers2[0] = strategiesAndWeightingMultipliers[strategiesAndWeightingMultipliers.length - 1];
+
+        // create quorum with all but the last element
+        uint8 quorumNumber = voteWeigher.quorumCount();
+        cheats.startPrank(serviceManagerOwner);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers1);
+
+        // add the last element
+        cheats.expectRevert("VoteWeigherBase._addStrategiesConsideredAndMultipliers: quorumNumber exceeds quorumCount");
+        voteWeigher.addStrategiesConsideredAndMultipliers(quorumNumber+1, strategiesAndWeightingMultipliers2);        
+    }
+
+    function testRemoveStrategiesConsideredAndMultipliers_Valid(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers,
+        uint256[] memory indicesToRemove
```

Comment:
this is a complex (at least in terms of implementation) test. probably merits some top-level comment explaining it a bit more?

Reply:
overall, I feel like there *has* to be a much more efficient way to implement this, but I feel like I can follow your reasoning

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/53/commits/7992f656918f7852cf5c9b5e7c383fb2241a1313